data/txHandler.go

```diff
@@ -328,19 +414,134 @@ type SolicitedTxHandler interface {
+		backlogGroups:          make(chan *txGroups, txBacklogSize),
+		skipNextBacklogWarning: false,
+	}
+}
+
+// HandleTransactionGroups implements the SolicitedAsyncTxHandler.HandleTransactionGroups interface
+// HandleTransactionGroups enqueues the given slice of transaction groups that came from the given network peer with
+// the given message sequence number. The provided acknowledgement channel provides a feedback for the transaction sync
+// that the entire transaction group slice was added ( or already included ) within the transaction pool. The method
+// return true if it's able to enqueue the processing task, or false if it's unable to enqueue the processing task.
+func (handler *solicitedAyncTxHandler) HandleTransactionGroups(networkPeer interface{}, ackCh chan uint64, messageSeq uint64, groups []transactions.SignedTxGroup) (enqueued bool) {
+	select {
+	case handler.backlogGroups <- &txGroups{networkPeer: networkPeer, txGroups: groups, ackCh: ackCh, messageSeq: messageSeq}:
+		// reset the skipNextBacklogWarning once the number of pending items on the backlogGroups channels goes to
+		// less than half of it's capacity.
+		if handler.skipNextBacklogWarning && (len(handler.backlogGroups)*2 < cap(handler.backlogGroups)) {
+			handler.skipNextBacklogWarning = false
+		}
+		enqueued = true
+	default:
+		if !handler.skipNextBacklogWarning {
+			logging.Base().Warnf("solicitedAyncTxHandler exhusted groups backlog")
+			handler.skipNextBacklogWarning = true
+		}
+		// if we failed here we want to increase the corresponding metric. It might suggest that we
+		// want to increase the queue size.
+		transactionMessagesDroppedFromBacklog.Inc(nil)
+	}
+	return
+}
+
+func (handler *solicitedAyncTxHandler) Start() {
+	if handler.stopCtxFunc == nil {
+		var ctx context.Context
+		ctx, handler.stopCtxFunc = context.WithCancel(context.Background())
+		handler.stopped.Add(1)
+		go handler.loop(ctx)
+	}
+}
+
+func (handler *solicitedAyncTxHandler) Stop() {
+	if handler.stopCtxFunc != nil {
+		handler.stopCtxFunc()
+		handler.stopped.Wait()
+		handler.stopCtxFunc = nil
+	}
+}
+func (handler *solicitedAyncTxHandler) loop(ctx context.Context) {
+	defer handler.stopped.Done()
+	var groups *txGroups
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case groups = <-handler.backlogGroups:
+		}
+		disconnect, allTransactionsIncluded := handler.txHandler.processDecodedArray(groups.txGroups)
+		if disconnect {
+			handler.txHandler.net.Disconnect(groups.networkPeer)
+			handler.txHandler.net.RequestConnectOutgoing(false, make(chan struct{}))
+			transactionMessagesDroppedFromPool.Inc(nil)
+		} else if allTransactionsIncluded {
+			select {
+			case groups.ackCh <- groups.messageSeq:
+				// all good, write was successful.
+			default:
+				// unable to write since channel was full - log this:
+				logging.Base().Warnf("solicitedAyncTxHandler was unable to ack transaction groups inclusion since the acknowledgement channel was full")
```

Comment:
Why do we want to drop the acknowledgement instead of blocking here?

Reply:
blocking here means that we won't be processing the incoming transaction groups and wasting precious computing time on waiting. this would lead to the incoming queue getting full, and incoming messages would not be processed. Instead, it would be a better choice to skip the ack, and keep processing the incoming messages.

Keep in mind that there always will be a case where we receive too many messages - especially for relays. In these cases we need to make the hard decisions ( i.e. no winners in either case ). In this case, I tried to ensure that we won't loose any incoming message over letting the other party know that we received a particular message. The other party might or might not send these transactions again in the future, but we might not have that load volume then.